crates/kitsune_p2p/proxy/src/tx2.rs

```diff
@@ -851,8 +854,11 @@ impl ProxyEp {
                             break;
                         }
 
-                        let timeout = tuning_params.implicit_timeout();
-                        let _ = hnd.get_connection(proxy_url.clone(), timeout).await;
+                        if let Some(proxy_url) = client_of_remote_proxy.get_proxy_url().await {
```

Comment:
Is it intended to query the bootstrap server here?

Reply:
Correct, `get_proxy_url()` will do the bootstrap query.

Reply:
As we get more mature, we can do fancy things within there like pinging all the proxies that were returned by the query and selecting the one with the lowest latency.